executor/api/util/utils.go

```diff
@@ -94,6 +95,34 @@ func ExtractRouteFromSeldonMessage(msg *proto.SeldonMessage) []int {
 	return []int{-1}
 }
 
+func InsertRouteToSeldonPredictPayload(msg payload.SeldonPayload, routing *map[string]int32) (payload.SeldonPayload, error) {
+
+	if msg.GetContentType() == payload.APPLICATION_TYPE_PROTOBUF {
+        sm := msg.GetPayload().(*proto.SeldonMessage)
+        sm.Meta.Routing = *routing
```

Comment:
I thought protos were read only - so interesting this works.

Reply:
Oh ok, I didn't know this, it does seem to work from the tests, so I can double check but all seems good on this imlpementation